pkg/controllers/disruption/queue.go

```diff
@@ -228,7 +230,11 @@ func (q *Queue) waitOrTerminate(ctx context.Context, cmd *Command) (err error) {
 	errs := make([]error, len(cmd.Candidates))
 	workqueue.ParallelizeUntil(ctx, len(cmd.Candidates), len(cmd.Candidates), func(i int) {
 		if err := retry.OnError(retry.DefaultBackoff, func(err error) bool { return client.IgnoreNotFound(err) != nil }, func() error {
-			return q.kubeClient.Delete(ctx, cmd.Candidates[i].NodeClaim)
+			e := q.kubeClient.Delete(ctx, cmd.Candidates[i].NodeClaim)
+			if e != nil {
+				q.tracker.FinishCommand(ctx, cmd.Candidates[i].NodeClaim)
```

Comment:
Didn't we determine that we would consider a command finished when we finalized the node in the termination controller? Finalizing it here just means that we've successfully brought up the replacement capacity.